keyboards/kbd6x/keymaps/othi/keymap.c

```diff
@@ -0,0 +1,225 @@
+/* Copyright 2018 Othi
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+// read README.md for rundown of what does what
+
+// #define BL 0
+#define CL 0
+#define NM_MODE 2
+#define VI_MODE 3
+#define FN3 4
+#define ACCENT 5
+#define ACCENT_CAP 6
+enum custom_keycodes {
+    MY_CUSTOM_MACRO = SAFE_RANGE,
+    DE_AE, /* Ctrl + Shift + u00E4 */
+    DE_SS, /* Ctrl + Shift + u00DF */
+    DE_OE, /* Ctrl + Shift + u00F6 */
+    DE_UE, /* Ctrl + Shift + u00FC */
+    FLUSH, /* clears oneshots */
+    DE_AE_CAP, /* Ctrl + Shift + u00C4 */
+    DE_OE_CAP, /* Ctrl + Shift + u00D6 */
+    DE_UE_CAP, /* Ctrl + Shift + u00DC */
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    if (record->event.pressed) {
+        switch(keycode) {
+            case MY_CUSTOM_MACRO:
+                SEND_STRING("QMK is the best thing ever!"); // this is our macro!
+                return false;
+            case DE_AE:
```

Comment:
If this is to send unicode characters (which it appears to be so), you may want to just enable unicode, as this changes this much better. 

https://docs.qmk.fm/#/feature_unicode

Eg, you could just use `UC(0x00E4)` in your keymap. 

Or even `send_unicode_hex_string`
https://docs.qmk.fm/#/feature_unicode?id=send_unicode_hex_string